pkg/imgpkg/bundle/bundle.go

```diff
@@ -142,12 +158,21 @@ func (o *Bundle) pull(baseOutputPath string, ui goui.UI, pullNestedBundles bool,
 		return err
 	}
 
-	localizedImagesLockToRepo, notLocalizedToBundle, err := NewImagesLock(imagesLock, o.imgRetriever, o.Repo()).LocalizeImagesLock()
+	lock := NewImagesLock(imagesLock, o.imgRetriever, o.Repo(), locationFetcher)
+	localizedImagesLockToRepo, notLocalizedToBundle, err := lock.LocalizeImagesLock()
 	if err != nil {
 		return err
 	}
 
 	if pullNestedBundles {
+		imageBundles := map[string]bool{}
+		locationsConfig, err := locationFetcher.Fetch()
```

Comment:
it seems that we are dealing with the guts of ImagesLock class, but outside of it? it would be better we encapsulate it into ImagesLock.